helper/keysutil/policy.go

```diff
@@ -431,14 +434,14 @@ func (p *Policy) Upgrade(storage logical.Storage) error {
 		persistNeeded = true
 	}
 
-	if p.Keys[p.LatestVersion].HMACKey == nil || len(p.Keys[p.LatestVersion].HMACKey) == 0 {
-		entry := p.Keys[p.LatestVersion]
+	if p.Keys[strconv.Itoa(p.LatestVersion)].HMACKey == nil || len(p.Keys[strconv.Itoa(p.LatestVersion)].HMACKey) == 0 {
```

Comment:
Same here, set `lastestVersion := strconv.Itoa(p.LatestVersion)` so you don't have to call it multiple times.